### PR TITLE
Add GPIO pads for UART and SPI Flash

### PR DESCRIPTION
**Pull request by IanJiangICT**
Add GPIO pads for UART2/3 and Flash ast SPI0.

### CODE REVIEW

File: arch/riscv/mach-duowen/gpio.c [hunk 1]
```diff
@@ -54,6 +54,14 @@ struct duowen_pad duowen_pads[] = {
...
 #endif
+#ifndef CONFIG_DUOWEN_FLASH_SPI0
+	/* flash0 */
+	{ GPIO1B, pad_gpio_48, TLMM_PAD_FUNCTION, GPIO_PAD_PULL_UP, 8 },
+	{ GPIO1B, pad_gpio_49, TLMM_PAD_FUNCTION, GPIO_PAD_PULL_UP, 8 },
+	{ GPIO1B, pad_gpio_50, TLMM_PAD_FUNCTION, GPIO_PAD_PULL_UP, 8 },
+	{ GPIO1B, pad_gpio_51, TLMM_PAD_FUNCTION, GPIO_PAD_PULL_UP, 8 },
+	{ GPIO1B, pad_gpio_52, TLMM_PAD_FUNCTION, GPIO_PAD_PULL_UP, 8 },
+#endif
 };
```
Comment (zetalog) [inline arch/riscv/mach-duowen/gpio.c:65]: This configuration looks wrong, please refer to
arch/riscv/mach-duowen/ssi_flash.c.

And you can also refer to arch/riscv/include/asm/mach-duowen/spi.h to add
ifndef CONFIG_DUOWEN_SSI_FLASH_SPI1/2/3
related configration.

File: arch/riscv/mach-duowen/gpio.c [hunk 2]
```diff
@@ -54,6 +54,14 @@ struct duowen_pad duowen_pads[] = {
 	{ GPIO2A, pad_gpio_108, TLMM_PAD_FUNCTION, GPIO_PAD_PULL_UP, 8 },
 	{ GPIO2A, pad_gpio_109, TLMM_PAD_FUNCTION, GPIO_PAD_PULL_UP, 8 },
 #endif
+#ifndef CONFIG_DUOWEN_FLASH_SPI0
```
Comment (zetalog) [inline arch/riscv/mach-duowen/gpio.c:57]: Should it be CONFIG_DUOWEN_SSI_FLASH_SPI0?

File: arch/riscv/mach-duowen/gpio.c (hunk 1 above)
Comment (IanJiangICT) [reply arch/riscv/mach-duowen/gpio.c:65]: fixed

File: arch/riscv/mach-duowen/gpio.c (hunk 2 above)
Comment (IanJiangICT) [reply arch/riscv/mach-duowen/gpio.c]: fixed

File: arch/riscv/mach-duowen/Kconfig [hunk 3]
```diff
@@ -645,6 +645,10 @@ config DUOWEN_SSI_FLASH_IRQ
 
 endif
 
+config DUOWEN_SSI_FLASH_DTS
+	bool "Enable dw-apb-ssi driver in DTS"
+	depends FDT
+
 menuconfig DUOWEN_SD
```
Comment (zetalog) [inline arch/riscv/mach-duowen/Kconfig:652]: Should it be DUOWEN_SSI_MASTER_DTS, and be put between if DUOWEN_SSI_FLASH/endif?

Comment (IanJiangICT) [reply arch/riscv/mach-duowen/Kconfig:652]: done

**Pull request merged**